File: network/src/main/java/com/difft/android/websocket/internal/push/PreKeyState.java
package com.difft.android.websocket.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import org.signal.libsignal.protocol.IdentityKey;
import com.difft.android.websocket.internal.util.JsonUtil;

public class PreKeyState {

    @JsonProperty
    @JsonSerialize(using = JsonUtil.IdentityKeySerializer.class)
    @JsonDeserialize(using = JsonUtil.IdentityKeyDeserializer.class)
    private IdentityKey identityKey;
    @JsonProperty
    private String newSign;

    public PreKeyState() {
    }

    public PreKeyState(IdentityKey identityKey, String identityKeySign) {
        this.identityKey = identityKey;
        this.newSign = identityKeySign;
    }

    public IdentityKey getIdentityKey() {
        return identityKey;
    }

    public String getNewSign() {
        return newSign;
    }
}

File: network/src/main/java/com/difft/android/websocket/internal/push/PreKeyState.kt
package com.difft.android.websocket.internal.push

import com.difft.android.websocket.internal.util.JsonUtil
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.signal.libsignal.protocol.IdentityKey

class PreKeyState {

    @field:JsonProperty
    @field:JsonSerialize(using = JsonUtil.IdentityKeySerializer::class)
    @field:JsonDeserialize(using = JsonUtil.IdentityKeyDeserializer::class)
    var identityKey: IdentityKey? = null
        private set

    @field:JsonProperty
    var newSign: String? = null
        private set

    constructor()

    constructor(identityKey: IdentityKey?, identityKeySign: String?) {
        this.identityKey = identityKey
        this.newSign = identityKeySign
    }
}
